docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,387 @@
+  * Q4.2: (Depends on answer for Q3) If no: What Presentation Request format need to be supported? DIF? 
+* Q5: Do we want to provide general interfaces doing signing and verification of Data Integrity proof signature?
+  * Accept sign/verify callbacks into convert functions:
+    * Issue with setting multiple signatures
+    * Much easier just to expose methods to add signature proof itself 
+* Q6: Signature/Proof encoding: Which approach to use for encoding? 
+  * Basic approach used in Aries attachments - Base 64 encoding of serialized object (`base64(json(signature).as_bytes())`) ?
+  * Compact encoding implementing in Python PoC?
+
+### Proposed implementation path for first iteration
+
+1. Credential conversion APIs
+2. Credential methods (set_integrity_proof, get_signing_payload?, etc?) - do not care about Data Integrity proof signatures
+3. Flow duplication APIs
+4. Adopt Credential Offer and Credential Request for W3C standard ????
+5. Support only Indy presentation request type (for the first iteration)
+
+### Public API
+
+#### Credential/Presentation Conversion methods
+
+The idea for this approach is only provide conversion method for credentials and presentations.
+So credentials and presentations themselves are generate the same way and the same functions as before but if W3C form is require application uses conversion methods to get required format.
+
+#### Credential Conversion methods
+
+These methods allow to solve both cases: 
+* Use old Indy styled credentials issued before 
+* Issuing new credentials in W3C form
+
+```rust
+/// Convert Indy styled AnonCreds credential into W3C AnonCreds credential form
+///     The conversion process described at the specification: ---
+///
+/// # Params
+/// cred -      object handle pointing to Indy styled credential to convert
+/// cred_p -    reference that will contain converted credential (in W3C form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_credential_to_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+
+/// Convert W3C styled AnonCreds credential into Indy styled credential
+///     The conversion process described at the specification: ---
+/// 
+/// # Params
+/// cred -      object handle pointing to W3C styled AnonCreds credential to convert
+/// cred_p -    reference that will contain converted credential (in Indy form) instance pointer
+///
+/// # Returns
+/// Error code
+#[no_mangle]
+pub extern "C" fn anoncreds_credential_from_w3c(
+    cred: ObjectHandle,
+    cred_p: *mut ObjectHandle,
+) -> ErrorCode {}
+```
+
+#### Credential object methods
+
+```rust
+/// Get W3C credential payload to sign for making Identity Proof
+///
+/// # Params
+/// cred -                  object handle pointing to W3C styled AnonCreds credential
+/// signing_payload_p -     reference that will contain payload to sign for generation of data integrity proof 
```

Comment:
I am really not too familiar with the data integrity proof, but I don't see the creation of the input that will be signed for a data integrity proof to be the responsibility of this library, right? Might be very wrong here, so please correct me if that is the case.

Reply:
Or will the CL sig be a data integrity proof?

Reply:
You are right. I'll remove this method. It will be the responsibility of 3td party library to create/verify non-AnonCreds identity proof signature.